### DISCUSSION TITLE
How to avoid NotSupportedException when specifing a port?

### DISCUSSION THREAD

**Discussion by redwoodsquirrel** [dotnet/MQTTnet#1771] [Q&A] [answered]
Hi all,

I have the following code snipped to connect to my broker:
```
var mqttClientOptions = new MqttClientOptionsBuilder()
	.WithTcpServer( "localhost", 5672 )
	.WithProtocolVersion( MqttProtocolVersion.V311 )
	.WithCredentials( Username, Password )
	.Build();

await mqttClient.ConnectAsync( mqttClientOptions, cancellationToken ).ConfigureAwait( false );
```

When I use the `.WithTcpServer( BrokerUrl, BrokerPort )` method I run into a NotSupportedException. ('Error while authenticating. Extended authentication handler is not yet supported') thrown in line 447 of the MqttClient. When I go with the `.WithTcpServer( BrokerUrl )` method this does not happen. I am using MQTTnet 4.2.1.781 with .NET6.

**My question is: Am I able to specify a port in some way or another, or is this not necessary at all?**
In my current scenario I have a MQTT broker running in a Docker container.

Thank you very much in advance!

**Comment by chkr1011** [comment]
If you avoid the port the default one (1883) will be used which seems to not require the extended authentication (which is not yet supported in the library). 

So your brokers seems to expose both ports right now.

**Reply by redwoodsquirrel** [reply]
Thanks for the quick response.
Please let me ask two follow up questions:
- What exactly is the ExtendentAuthentication besides defining a specific port?
- Is there a time estimate when this might be available in the future?

**Reply by chkr1011** [reply, marked as answer]
Extended authentication is not related to ports in any way. It is a feature of MQTTv5 which might be used by your broker. But it turns out that it may not be the root cause and the message is misleading. Please try again with the next release version which will be released soon. It provides more details about what is happening.
